### ISSUE TITLE
Vibrate fails on iOS using custom pattern

### ISSUE THREAD

**Issue by flyingcircle** [benjamindean/flutter_vibration#79]
iOS 16.1.2
iPhone 11

I currently run the following code:
```dart
Vibration.vibrate(pattern: [500, 500, 500, 500]);
```

```
[hapi]  CHHapticPatternPlayer.mm:132   -[PatternPlayer initWithPlayable:engine:privileged:error:]: ERROR: pattern contains no playable events
Failed to play pattern: The operation couldn’t be completed. (Foundation._GenericObjCError error 0.).
```

Both `Vibration.hasVibrator()` and `Vibration.hasCustomVibrationsSupport()` return `true`. So this appears like it should work, but instead error is being thrown.

**Comment by oorfalimak** [comment]
I have the same problem, too.

Iphone SE(2020)
iOS 13.4.1

**Comment by NicklasFrederiksen** [comment]
Same problem

**Comment by n04h** [comment]
I also have the same problem.

iPhone 11 Pro
iOS 16.2

**Comment by Amareis** [comment]
Ok, looking at code in https://github.com/benjamindean/flutter_vibration/blob/master/vibration/ios/Classes/VibrationPluginSwift.swift#L81 it looks like you NEED to pass amplitudes, there is no any default values for them -_-

**Comment by Amareis** [comment]
It was broken in https://github.com/benjamindean/flutter_vibration/pull/51

**Comment by flyingcircle** [comment]
@Amareis that fixed the issue. If someone wants to take a stab at the PR for a default value that would fix iOS.

**Comment by Amareis** [comment]
Yes, and also simple `vibrate(duration: duration)` can be emulated via `vibrate(pattern: [0, duration], intensities: [0, 255])`. I think it should be handled in swift plugin directly, instead of fallback to dumb long standart vibration.

**Comment by pranavo72bex** [comment]
Vibration.vibrate(
        pattern: [500, 1000, 500, 2000, 500, 3000, 500, 500],
        repeat: 1,
        intensities: [0, 128, 0, 255, 0, 64, 0, 255],
      );
      This code is not working in iOS devices.

**Comment by bibek-magar** [comment]
> Vibration.vibrate( pattern: [500, 1000, 500, 2000, 500, 3000, 500, 500], repeat: 1, intensities: [0, 128, 0, 255, 0, 64, 0, 255], ); This code is not working in iOS devices.

Its not working on mine too..

**Comment by Amareis** [comment]
Try it without repeat

**Comment by pranavo72bex** [comment]
> Try it without repeat

so how to make iOS keep device vibrating continuously?

**Comment by pranavo72bex** [comment]
@Amareis  i am getting same error 
```
[hapi]  CHHapticPatternPlayer.mm:132   -[PatternPlayer initWithPlayable:engine:privileged:error:]: ERROR: pattern contains no playable events
Failed to play pattern: The operation couldn’t be completed. (Foundation._GenericObjCError error 0.).
```

**Comment by pranavo72bex** [comment]
any update?

**Comment by pranavo72bex** [comment]
@flyingcircle how did you manage to fix the issue?

**Comment by benjamindean** [comment]
Hey! Sorry for the late reply, but I will soon get an actual iPhone to debug it on so I'll look into it. Thank you for your patience!

**Comment by luomo-pro** [comment]
Is there any progress?

**Comment by benjamindean** [comment]
I released the new version with fixed duration, pattern and intensity handling. I tested it out manually and all seems to work fine. You can now change duration of vibration and provide vibration without pattern and intensities on iOS. Let me know if you have any other problems and sorry for the long wait!
